lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -458,11 +463,87 @@ bool SessionState::apply_update_criteria(SessionStateUpdateCriteria& uc) {
   return true;
 }
 
+RuleStats SessionState::get_rule_delta(
+    const std::string& rule_id, uint64_t rule_version, uint64_t used_tx,
+    uint64_t used_rx, uint64_t dropped_tx, uint64_t dropped_rx,
+    SessionStateUpdateCriteria& uc) {
+  RuleStats ret = {0};
+
+  auto it = policy_version_and_stats_.find(rule_id);
+  if (it == policy_version_and_stats_.end()) {
+    MLOG(MERROR) << "Reported rule not found, ignoring";
+    return ret;
+  }
+
+  // Only accept rule reports for current_version or last_reported_version
+  // ignore other reports as they shoudn't be sent
+  auto last_reported_version = it->second.last_reported_version;
+  if (rule_version > it->second.current_version) {
+    MLOG(MERROR) << "Reported version higher than tracked one("
+                 << it->second.current_version << ") for "
+                 << "Rule_id: " << rule_id << " version: " << rule_version;
+    return ret;
+  }
+
+  if (rule_version < last_reported_version) {
+    MLOG(MERROR) << "Reported rule version too old, current one("
+                 << it->second.current_version << ") for "
+                 << "Rule_id: " << rule_id << " version: " << rule_version;
+    return ret;
+  }
+
+  auto last_tracked =
+      policy_version_and_stats_[rule_id].stats_map.find(last_reported_version);
+
+  RuleStats prev_usage = last_tracked->second;
+
+  if (rule_version == last_reported_version) {
+    if (prev_usage.tx != 0 && prev_usage.tx > used_tx) {
+      MLOG(MERROR) << "Reported stat used_tx less than the current tracked one";
```

Comment:
Should we just keep one of the values (old/new) as a best effort estimation?
Maybe just always take the higher value and log at a warning level?

Reply:
I'll bump the log to warn. Its already keeping the highest value (prev_usage.tx > used_tx would only trigger if new report is showing less stats)

Reply:
oh gotcha! will read again haha :D